source/Engine/Rendering/GL/GLRenderer.cpp

```diff
@@ -2099,53 +2099,66 @@ void     GLRenderer::DeleteVertexBuffer(void* vtxBuf) {
 
     delete vertexBuffer;
 }
-void     GLRenderer::MakeFrameBufferID(ISprite* sprite, AnimFrame* frame) {
-    frame->ID = 0;
-
+void     GLRenderer::MakeFrameBufferID(ISprite* sprite) {
     float fX[4], fY[4];
     fX[0] = 1.0;    fY[0] = 1.0;
     fX[1] = -1.0;   fY[1] = 1.0;
     fX[2] = 1.0;    fY[2] = -1.0;
     fX[3] = -1.0;   fY[3] = -1.0;
 
     GL_AnimFrameVert vertices[16];
-    GL_AnimFrameVert* vert = &vertices[0];
 
-    if (frame->SheetNumber >= sprite->Spritesheets.size())
-        return;
-    if (!sprite->Spritesheets[frame->SheetNumber])
-        return;
+    if (sprite->ID == 0) {
+        glGenBuffers(1, (GLuint*)&sprite->ID); CHECK_GL();
+    }
+    glBindBuffer(GL_ARRAY_BUFFER, sprite->ID); CHECK_GL();
+    glBufferData(GL_ARRAY_BUFFER, sizeof(vertices) * sprite->FrameCount, NULL, GL_STATIC_DRAW); CHECK_GL();
+
+    size_t fc = 0;
+    for (size_t a = 0; a < sprite->Animations.size(); ++a) {
+        for (size_t i = 0; i < sprite->Animations[a].Frames.size(); ++i) {
+            AnimFrame* frame = &sprite->Animations[a].Frames[i];
+            GL_AnimFrameVert* vert = &vertices[0];
+
+            if (frame->SheetNumber >= sprite->Spritesheets.size())
+                continue;
+            if (!sprite->Spritesheets[frame->SheetNumber])
+                continue;
```

Comment:
Not needed as those entries can't be `null`

Reply:
Actually, I just noticed this was in the original code... My point still stands, though.

Reply:
Sounds good, I didn't know what this check was for either honestly enough LOL